tests/rbd_mirror/test_mirroring_toggle_journaling_feature.py

```diff
@@ -0,0 +1,217 @@
+import json
+from time import sleep
+
+from ceph.parallel import parallel
+from ceph.rbd.initial_config import initial_mirror_config
+from ceph.rbd.utils import getdict
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.rbd_mirror import wait_for_replay_complete
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def create_delete_snapshot(rbd, pool, image, operation, count):
+    """ """
+    for i in range(0, count):
+        snap_spec = {"snap-spec": f"{pool}/{image}@{image}_snap_{i}"}
+        if operation == "create":
+            out, err = rbd.snap.create(**snap_spec)
+        else:
+            out, err = rbd.snap.rm(**snap_spec)
+        if out or err and "100% complete" not in err:
+            raise Exception(f"Snapshot {operation} failed for {snap_spec}")
+    return 0
+
+
+def toggle_journaling_and_verify(rbd1, rbd2, pool, image):
+    """ """
+    sleep(15)
+    feature_spec = {"image-spec": f"{pool}/{image}", "features": "journaling"}
+    out, err = rbd1.feature.disable(**feature_spec)
+    if out or err:
+        raise Exception(f"Journaling feature disable failed for {image}")
+
+    image_spec = {"image-or-snap-spec": f"{pool}/{image}"}
+
+    out, err = rbd2.info(**image_spec)
+
+    if out or err:
+        log.info(
+            "Image not present in secondary when journaling is disabled in primary"
+        )
+    else:
+        raise Exception(
+            "Image present in secondary even though journaling is disabled in primary"
+        )
+
+    sleep(15)
+
+    out, err = rbd1.feature.enable(**feature_spec)
+    if out or err:
+        raise Exception(f"Journaling feature enable failed for {image}")
+
+    sleep(15)
```

Comment:
why sleep here?

Reply:
To wait for feature enable to reflect before checking mirroring status on secondary. Since when feature is disabled image gets deleted in secondary and when we enable we should wait for secondary image to get recreated before checking for status.